actor/src/main/scala-2.12/org/apache/pekko/util/ByteString.scala

```diff
@@ -433,6 +447,20 @@ object ByteString {
       }
     }
 
+    override def indexOf(elem: Byte): Int = indexOf(elem, 0)
+    override def indexOf(elem: Byte, from: Int): Int = {
+      if (from >= length) -1
+      else {
+        var found = -1
+        var i = math.max(from, 0)
+        while (i < length && found == -1) {
```

Comment:
I think if we swap with `found == -1 && i < length)` which can reduce a bit when we just found:)